web/src/containers/IdleLogout.js

```diff
@@ -16,26 +17,97 @@ const events = [
   'scroll'
 ];
 
-const IDLE_TIMEOUT = process.env.IDLE_LOGOUT_TIME_MINUTES * 60 * 1000;
+// const IDLE_TIMEOUT = process.env.IDLE_LOGOUT_TIME_MINUTES * 60 * 1000;
+const IDLE_TIMEOUT = 2.2 * 60 * 1000;
+
+const flashWindowTitle = () => {
+  const newTitle = `🔴 ${document.title}`;
+  const originalTitle = document.title;
+  const timer = setInterval(() => {
+    document.title =
+      document.title === originalTitle ? newTitle : originalTitle;
+  }, 300);
+
+  return () => {
+    clearInterval(timer);
+    document.title = originalTitle;
+  };
+};
 
 const IdleLogout = ({ history, loggedIn }) => {
   useEffect(() => {
     if (loggedIn) {
+      let notification = null;
+      let stopFlashing = () => {};
+
+      if (window.Notification) {
+        window.Notification.requestPermission();
+      }
+
       const logout = () => {
         history.replace('/logout');
       };
 
+      const idleWarning = () => {
+        stopFlashing = flashWindowTitle();
+
+        if (window.Notification) {
+          notification = new Notification('Session expiring', {
+            body: 'You will be logged out soon',
+            requireInteraction: true
+          });
+          notification.onclick = () => {
+            resetTimer(); // eslint-disable-line no-use-before-define
```

Comment:
Normally I like this lint rule that you can't use things that the reader hasn't seen before, but it's tough to figure out how to break that in a circular dependency situation.